server/controllers/admin.py

```diff
@@ -1043,6 +1043,24 @@ def client(client_id):
         return redirect(url_for(".clients"))
     return render_template('staff/edit_client.html', client=client, form=form, courses=courses)
 
+@admin.route("/course/<int:cid>/unenrolled", methods=['GET'])
+@is_staff(course_arg='cid')
+def unenrolled(cid):
+    courses, current_course = get_courses(cid)
+    
+    submissions = set(b.submitter_id for b in (Backup.query.join(Backup.assignment).filter(Assignment.course_id == cid).all()))
+    enrollment = set(e.user_id for e in (Enrollment.query.filter(Enrollment.course_id == cid).all()))
+    unenrolled_submitters = []
+    
+    for s in submissions:
+        if s not in enrollment:
+            unenrolled_submitters.append(User.query.get(s))
```

Comment:
See above - you should be able to do e.user if you use enrollment objects directly